addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -70,17 +70,38 @@ weSnippetEditor.Class.include({
                 size: 'medium',
                 title: _t("Google Map API Key"),
                 buttons: [
-                    {text: _t("Save"), classes: 'btn-primary', close: true, click: async () => {
-                        const newAPIKey = dialog.$('#api_key_input').val() || false;
-                        await this._rpc({
-                            model: 'website',
-                            method: 'write',
-                            args: [
-                                [websiteId],
-                                {google_maps_api_key: newAPIKey},
-                            ],
-                        });
-                        invalidated = true;
+                    {text: _t("Save"), classes: 'btn-primary', close: false, click: async (event) => {
+                        const inputApiKey = dialog.$('#api_key_input');
+                        const valueAPIKey = inputApiKey.val() || false;
+                        if (!valueAPIKey) {
+                            inputApiKey.addClass('is-invalid');
+                            dialog.$('#api_key_help').text(_t('Enter an API Key'));
+                            return;
+                        }
+                        let button = $(event.currentTarget);
```

Comment:
Detail but use "const" instead of "let" when you can. And again name that variable by starting with `$`